core/ledger/kvledger/kv_ledger_provider.go

```diff
@@ -252,6 +245,27 @@ func (p *Provider) initLedgerStatistics() {
 	p.stats = newStats(p.initializer.MetricsProvider)
 }
 
+func (p *Provider) initSnapshotDir() error {
+	snapshotsRootDir := p.initializer.Config.SnapshotsConfig.SnapshotsRootDir
+	if !path.IsAbs(snapshotsRootDir) {
+		return errors.Errorf("invalid path: %s. The path for the snapshot dir is expected to be an absolute path", snapshotsRootDir)
+	}
+
+	snapshotsTempDir := SnapshotsTempDir(snapshotsRootDir)
+	snapshotsFinalDir := SnapshotsFinalDir(snapshotsRootDir)
+
+	if err := os.RemoveAll(snapshotsTempDir); err != nil {
+		return errors.Wrapf(err, "error while deleting the dir: %s", snapshotsTempDir)
+	}
```

Comment:
Not sure that I understand this comment

Reply:
I thought we would create a temp file per store and once it is completed, we would rename it and move to other stores. If the peer starts after a failure, it would see which stores have been exported completely and which are pending. Then, it would skip all completed stores and export data out of pending store. I had this assumption when you mentioned about temp files and renaming it atomically. In the above comment, I meant to say that for now, we can go with the simple approach that this PR introduced.